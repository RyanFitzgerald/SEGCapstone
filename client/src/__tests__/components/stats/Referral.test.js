import React from 'react';
import Referral from '../../../Components/stats/Referral';

describe('<Referral />', () => {

  beforeAll(() => {
    // Add necessary info to user
    const user = {
      role: {
        level: 1
      }
    };
    localStorage.setItem('user', JSON.stringify(user));

  });

  it('should render correctly', () => {
    // Return empty array on requests
    mock.onAny().reply(200, []);
    // Create mock functions / spies
    const setActiveSubtabMock = jest.fn();
    const renderErrorMock = jest.fn();
    const getTotalVolumeByReferralSpy = jest.spyOn(Referral.prototype, 'getTotalVolumeByReferral');

    // Shallow render it
    const wrapper = shallow(
      <Referral setActiveSubtab={setActiveSubtabMock} renderError={renderErrorMock}/>
    );

    // Compare to old snapshot
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    // Check set active subtab is called and with correct value
    expect(getTotalVolumeByReferralSpy).toHaveBeenCalledTimes(1);
    expect(setActiveSubtabMock).toHaveBeenCalledTimes(1);
    expect(setActiveSubtabMock.mock.calls[0][0]).toBe(4);
  });
});
