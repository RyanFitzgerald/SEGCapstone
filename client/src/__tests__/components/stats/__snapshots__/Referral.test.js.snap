// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Referral /> should render correctly 1`] = `<Loading />`;
